index.d.ts

```diff
@@ -39,11 +39,12 @@ declare module 'i18next' {
 type ObjectOrNever = TypeOptions['allowObjectInHTMLChildren'] extends true
   ? Record<string, unknown>
   : never;
-type ReactI18NextChild = React.ReactNode | ObjectOrNever;
+
+type ReactI18NextChildren = React.ReactNode | ObjectOrNever;
```

Comment:
Now, children will behave exactly like the [actual](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/v17/index.d.ts#L1342) unless `allowObjectInHTMLChildren` is `true`.